File: 4기-코딩테스트-220608/4기 코딩테스트_220608/4기 - 1st 코딩테스트/CheckColorCode/CheckColorCode_test.js
import assert from 'assert'
import solution from './CheckColorCode.js'

describe('Custom Test', function () {
  it('test 1', function (done) {
    // given
    const s = 'DACC32'

    // when
    const result = solution(s)

    // then
    assert.deepStrictEqual(true, result)
    done();
  });

  it('test 2', function (done) {
    // given
    const s = 'AAAAAA'

    // when
    const result = solution(s)

    // then
    assert.deepStrictEqual(true, result)
    done();
  });

  it('test 3', function (done) {
    // given
    const s = '000000'

    // when
    const result = solution(s)

    // then
    assert.deepStrictEqual(true, result)
    done();
  });

  it('test 4', function (done) {
    // given
    const s = 'A0B2C4'

    // when
    const result = solution(s)

    // then
    assert.deepStrictEqual(true, result)
    done();
  });

  it('test 5', function (done) {
    // given
    const s = 'ABAB'

    // when
    const result = solution(s)

    // then
    assert.deepStrictEqual(false, result)
    done();
  });

  it('test 6', function (done) {
    // given
    const s = '01234567'

    // when
    const result = solution(s)

    // then
    assert.deepStrictEqual(false, result)
    done();
  });
})

File: 4기 코딩테스트_220615/4기 - 1st 코딩테스트/CheckColorCode/CheckColorCode_test.py
import unittest

from src.com.luckydave.algorithm.fastcampus.hard.CheckColorCode.CheckColorCode import solution


class CustomTestCase(unittest.TestCase):

    def test_1(self):
        # given
        s = 'DACC32'

        # when
        result = solution(s)

        # then
        self.assertEqual(True, result)

    def test_2(self):
        # given
        s = 'AAAAAA'

        # when
        result = solution(s)

        # then
        self.assertEqual(True, result)

    def test_3(self):
        # given
        s = '000000'

        # when
        result = solution(s)

        # then
        self.assertEqual(True, result)

    def test_4(self):
        # given
        s = 'A0B2C4'

        # when
        result = solution(s)

        # then
        self.assertEqual(True, result)

    def test_5(self):
        # given
        s = 'ABAB'

        # when
        result = solution(s)

        # then
        self.assertEqual(False, result)

    def test_6(self):
        # given
        s = '01234567'

        # when
        result = solution(s)

        # then
        self.assertEqual(False, result)


if __name__ == '__main__':
    unittest.main()
